shared/src/store/action-reducers/simulated-region.ts

```diff
@@ -146,25 +237,7 @@ export namespace SimulatedRegionActionReducers {
         {
             action: RemoveSimulatedRegionAction,
             reducer: (draftState, { simulatedRegionId }) => {
-                const simulatedRegion = getElement(
-                    draftState,
-                    'simulatedRegion',
-                    simulatedRegionId
-                );
-                const transferPoint = getElementByPredicate(
-                    draftState,
-                    'transferPoint',
-                    (element) =>
-                        isInSpecificSimulatedRegion(element, simulatedRegion.id)
-                );
-                TransferPointActionReducers.removeTransferPoint.reducer(
-                    draftState,
-                    {
-                        type: '[TransferPoint] Remove TransferPoint',
-                        transferPointId: transferPoint.id,
-                    }
-                );
```

Comment:
The remove transfer point action reducer does more than just removing the transfer point, since it also handles existing transfers.

I think we should keep using its code (possibly by extracting all code from the reducer to a new function that can be called from the transfer point reducer as well as the simulated region reducer) and remove the transfer point before removing vehicles/materials/personnel, since removing the transfer point might cause elements to be added to the region.